ci/ci_run_all_tests.sh

```diff
@@ -39,6 +39,7 @@ dpkg -s mosquitto-clients
 python3 -m venv ~/env-pysys
 source ~/env-pysys/bin/activate
 pip3 install -r tests/requirements.txt
+sudo pip3 install retry
```

Comment:
Why sudo? The module needs elevated privileges? If not, could you add the the entry for `retry` in the `requirements.txt` file instead?

Reply:
The retry is used in create_test_logs. Which is called using `sudo`. So, this has to be installed using `sudo`.

Reply:
Hmm... That would lead to a weird Python dependency requirement with every other module installed into the virtual environment and this module going to the system default python (I assume). Don't think that's an ideal setup. I believe the problem here is that we're calling that script using the system default `python3` instead of the venv one we use for these tests. Can't that log generator python script be called from the tests using the Pysys `startProcess` API with `sudo` like we invoke other scripts?

Reply:
The current solution seems contrived, with a cascade of drawbacks. Why are the logs created using an external script launched as `sudo python3` ?

Reply:
To create the logs in /var/log/tedge/agent/, script has to be run as sudo. To run invoke a script without python3 the script has to set some environment and has to be made executable, by changing the permission.

Reply:
How about exposing a function from that script which is used in the test? To get around the sudo permission issue, what you can do is to write the generated log file to /tmp and then move it to `/var/log/tedge/agent` directory with `startProcess("sudo mv /tmp/log/path /var/log/tedge/agent")`.

Reply:
exactly implemented the same